openedx/adg/lms/webinars/admin.py

```diff
@@ -2,35 +2,86 @@
 Registering models for webinars app.
 """
 from django.contrib import admin
+from django.db.models import Q
+from django.utils.translation import ugettext_lazy as _
 
 from openedx.adg.lms.applications.admin import adg_admin_site
 
-from .models import Webinar, WebinarRegistration
+from .models import CancelledWebinar, Webinar, WebinarRegistration
 
 
-@admin.register(Webinar)
-class WebinarAdmin(admin.ModelAdmin):
+class ActiveWebinarStatusFilter(admin.SimpleListFilter):
     """
-    Model admin for webinar
+    Custom filter to provide `Upcoming` and `Delivered` states filter functionality to the WebinarAdmin
     """
 
-    save_as = True
+    title = _('Webinar Status')
+    parameter_name = 'status'
+
+    def lookups(self, request, model_admin):
+        return (
+            (Webinar.UPCOMING, _('Upcoming')),
+            (Webinar.DELIVERED, _('Delivered')),
+        )
+
+    def queryset(self, request, queryset):
+        if self.value() == Webinar.UPCOMING:
+            return queryset.filter(status=Webinar.UPCOMING)
+
+        if self.value() == Webinar.DELIVERED:
+            return queryset.filter(status=Webinar.DELIVERED)
+
+
+class WebinarAdminBase(admin.ModelAdmin):
+    """
+    Base Model admin for webinars i.e Cancelled Webinars and Non-Cancelled Webinars
+    """
 
     list_display = ('title', 'start_time', 'presenter', 'status',)
-    list_filter = ('start_time', 'status', 'language',)
     search_fields = ('title',)
-    readonly_fields = ('created_by', 'modified_by',)
     filter_horizontal = ('co_hosts', 'panelists',)
 
+
+class WebinarAdmin(WebinarAdminBase):
+    """
+    Admin for upcoming and delivered webinars
+    """
+
+    save_as = True
+    list_filter = ('start_time', 'language', ActiveWebinarStatusFilter)
+    readonly_fields = ('created_by', 'modified_by', 'status',)
```

Comment:
Making `status` read-only will impact work done in https://github.com/OmnipreneurshipAcademy/edx-platform/pull/249. For example updating status before cloning webinar.

Reply:
It will not affect the functionality we are trying to achieve under #249.
Any webinar we are trying to clone, despite its current status (upcoming or delivered); The resultant webinar will have the status set to `incoming`.

Great concern, but our functionality is working as it should, I double checked.